src/main/java/org/stellar/sdk/StrKey.java

```diff
@@ -49,6 +65,25 @@ public static String encodeStellarAccountId(AccountID accountID) {
         return String.valueOf(encoded);
     }
 
+    public static String encodeSignedPayload(SignedPayloadSigner signedPayloadSigner) {
+        if (signedPayloadSigner.getPayload().length > SIGNED_PAYLOAD_MAX_PAYLOAD_LENGTH) {
+            throw new FormatException("invalid payload length, must be less than " + SIGNED_PAYLOAD_MAX_PAYLOAD_LENGTH);
+        }
+        try {
+            ByteArrayOutputStream record = new ByteArrayOutputStream();
+            DataOutputStream dataStream = new DataOutputStream(record);
+            dataStream.write(signedPayloadSigner.getDecodedAccountId());
+            dataStream.writeInt(signedPayloadSigner.getPayload().length);
+            dataStream.write(signedPayloadSigner.getPayload());
+            int padding = signedPayloadSigner.getPayload().length % 4 > 0 ? 4 - signedPayloadSigner.getPayload().length % 4 : 0;
+            dataStream.write(new byte[padding]);
+            char[] encoded = encodeCheck(VersionByte.SIGNED_PAYLOAD, record.toByteArray());
+            return String.valueOf(encoded);
```

Comment:
Maybe we should make it explicit in SEP-23 that the signed payload specification matches the XDR. Would that help @Shaptic ?

Reply:
Yeah I think that was a big reason why I coded it the way I did initially: the description seemed like it was specifying a particular (and bespoke) binary packing, but in reality it was just a direct correspondence to `base32(xdr(signer) + xdr(payload))`. It would definitely help to tie the concepts together in the SEP, @leighmcculloch :+1:

Reply:
ah, good to know, I updated to use the xdr bindings for encoding, thanks!